### PR TITLE
fix(pypi): fix the whl selection algorithm after #2069

### PR DESCRIPTION
**Pull request by aignas**
It seems that a few things broke in recent commits:
- We are not using the `MODULE.bazel.lock` file and it seems that it is easy to
  miss when the components in the PyPI extension stop integrating well
  together. This happened during the switch to `{abi}_{os}_{plat}` target
  platform passing within the code.
- The logger code stopped working in the extension after the recent additions
  to add the `rule_name`.
- `repo_utils.getenv` was always getting `PATH` env var on bazel `6.x`.

This PR fixes both cases and updates docs to serve as a better reminder. By
fixing the `select_whls` code and we can just rely on target platform triples
(ABI, OS, CPU). This gets one step closer to maybe supporting optional
`python_version` which would address #1708. Whilst at it we are also adding
different status messages for building the wheel from `sdist` vs just
extracting or downloading the wheel.

Tests:
- Added more unit tests and brought them in line with the rest of the code.
- Checked manually for differences between the `MODULE.bazel.lock` files in our
  `rules_python` extension before #2069 and after this PR and there are no
  differences except in the `experimental_target_platforms` attribute in
  `whl_library`. Before this PR you would see that we do not select any wheels
  for e.g. `MarkupSafe` and we are always building from `sdist`.

Work towards #260.

### CODE REVIEW

File: python/private/repo_utils.bzl [hunk 1]
```diff
@@ -66,18 +67,25 @@ def _logger(rctx):
         "TRACE": 3,
     }.get(verbosity_level, 0)
 
+    if hasattr(ctx, "attr"):
+        # This is `repository_ctx`.
+        name = name or "{}(@@{})".format(getattr(ctx.attr, "_rule_name", "?"), ctx.name)
+    elif not name:
+        fail("The name has to be specified when using the logger with `module_ctx`")
+
+    name = getattr(ctx, "name", "extension")
```
Comment (rickeylev) [inline python/private/repo_utils.bzl:76]: This assignment is clobbering the logic above. I think you meant for this to be in the elif branch?

**Comment by rickeylev** [comment]
>  Before this PR you would see that we do not select any wheels for e.g. MarkupSafe and we are always building from sdist.

Would this show up as the "Fetching repo bla bla ..." step taking a long time? When I was working on the docs yesterday, there were a couple pypi deps (markup safe sounds familiar) that seemed to be taking a long time to become available. I chalked it up slow downloads or because I was modifying the pypi blz code, but maybe it was building it from source in the repo phase.

**Review by rickeylev — CHANGES_REQUESTED** [review]

**Comment by aignas** [comment]
Correct, it was building `MarkupSafe` from source, that is why it was taking longer than usual, I'll add a better log message to indicate that, because we can distinguish between that when we pass `url` to `whl_library`.

File: python/private/repo_utils.bzl (hunk 1 above)
Comment (aignas) [reply python/private/repo_utils.bzl]: Oops, this was leftover code, now removed.

**Review by rickeylev — APPROVED** [review]

**Pull request merged**